external/dhcpd/dhcpd_lwip.c

```diff
@@ -332,13 +332,14 @@ int dhcp_server_start(char *intf, dhcp_sta_joined_cb dhcp_join_cb)
 		return ERROR;
 	}
 	ret = pthread_create(&g_dhcpd_tid, &attr, _dhcpd_join_handler, (void *)data);
-	if (ret < 0) {
+	if (ret != OK) {
 		free(data->intf);
 		free(data);
 		ndbg("dhcps create dhcp handler fail(%d) errno %d\n", ret, errno);
 		return ERROR;
 	}
 	pthread_setname_np(g_dhcpd_tid, "dhcpd cb handler");
+	pthread_detach(g_dhcpd_tid);
```

Comment:
You don't need to call pthread_detach here. because pthread_join is called in dhcps_stop().